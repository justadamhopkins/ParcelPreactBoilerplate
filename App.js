import { h } from 'preact';
import { useQuery, useMutation, gql } from '@apollo/client';

export const bookQuery = gql`
	query {
		books {
			id
			title
		}
	}
`;

const mutation = gql`
	mutation {
		updateBookTitle(id: 1, title: "New title") {
			id
			title
		}
	}
`;

const App = () => {
	const { error, loading, data } = useQuery(bookQuery);
	const [update] = useMutation(mutation);

	if (loading) return <p>Loading...</p>;
	if (error) return <p>Error :(</p>;
	return (
		<div>
			<h1>{`Hello from Preact and the environment is ${process.env.ENVIRONMENT}`}</h1>
			<pre>{JSON.stringify(data, null, 2)}</pre>
			<button
				type='button'
				onClick={() => {
					update();
				}}
			>
				Update
			</button>
		</div>
	);
};

export default App;
