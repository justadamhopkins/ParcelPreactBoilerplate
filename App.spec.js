import { h } from 'preact';
import { render, cleanup } from '@testing-library/preact';
import '@testing-library/jest-dom';
import { MockedProvider } from '@apollo/client';
import App, { bookQuery } from './App';

describe('@App', () => {
	const mocks = [
		{
			request: {
				query: bookQuery,
			},
			result: {
				data: {
					books: [{ id: '1', title: 'HarryPotter' }],
				},
			},
		},
	];
	afterEach(cleanup);
	it('renders the app', () => {
		const { asFragment } = render(
			<MockedProvider mocks={mocks} addTypename={false}>
				<App />
			</MockedProvider>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
