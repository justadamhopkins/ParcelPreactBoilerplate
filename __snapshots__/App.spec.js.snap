// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@App renders the app 1`] = `
<DocumentFragment>
  <section
    class="App__Wrapper-ilpe0d-0 cldPfB"
  >
    <h1
      class="App__Title-ilpe0d-1 cNfSRS"
    >
      Hello from React and the environment is undefined
    </h1>
    <pre>
      {
  "books": [
    {
      "id": 1,
      "title": "HarryPotter"
    }
  ]
}
    </pre>
    <button
      class="App__Button-ilpe0d-2 fMSavx"
      type="button"
    >
      Update
    </button>
  </section>
</DocumentFragment>
`;
